File: web/src/main/java/net/danil/web/problem/model/SolutionResult.java
package net.danil.web.problem.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;

import java.util.Date;

@Table
@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class SolutionResult {
    @Id
    private Long id;

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private Date createdAt;

    @Column(nullable = false)
    private Integer tests;

    @Column(nullable = false)
    private Integer failures;

    @Column(nullable = false)
    private Integer errors;

    @Column(nullable = false)
    private Integer statusCode;

    @Column(nullable = false)
    private Float time;

    @Column(nullable = false)
    private String logs;

    private String junitXml;

    @Column(nullable = false)
    private Boolean solved;

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    private Solution solution;
}

File: web/src/main/java/net/danil/web/domain/SolutionResult.kt
package net.danil.web.domain

import jakarta.persistence.*
import net.danil.web.repository.TestListJsonConverter
import java.util.*


@Table(name = "solution_result")
@Entity
open class SolutionResult {
    @Id
    var id: Long? = null

    @Column(updatable = false)
    lateinit var createdAt: Date

    var exitCode: Int = 0
    var stdout: String = ""
    var stderr: String = ""

    var time: Float = 0f
    var timedOut: Boolean = false

    var solved: Boolean = false

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    lateinit var solution: Solution

    @Convert(converter = TestListJsonConverter::class)
    @Column(columnDefinition = "text")
    val tests: MutableList<TestResult> = ArrayList()
}